packages/epo-widget-lib/src/atomic/ElapsedTime/ElapsedTime.tsx

```diff
@@ -0,0 +1,57 @@
+import { FunctionComponent } from "react";
+import { Trans } from "react-i18next";
+import * as Styled from "./styles";
+
+type TimeInterval = "year" | "day" | "hour" | "minute" | "second";
+
+type TimeStepProps = {
+  [key in TimeInterval]?: number;
+};
+
+const ElapsedTime: FunctionComponent<TimeStepProps> = ({
+  year,
+  day,
+  hour,
+  minute,
+  second,
+}) => {
+  const steps = { year, day, hour, minute, second };
+  const validSteps = Object.keys(steps)
+    .filter((k) => steps[k as TimeInterval] !== undefined)
```

Comment:
filter out the numbers that weren't passed and create an array of intervals and their values